tests/test_tick_tock.sh

```
@@ -0,0 +1,260 @@
+TESTS_RUN=0
+TESTS_PASSED=0
+TESTS_FAILED=0
+
+# Get the directory where this script is located
+TEST_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(dirname "$TEST_DIR")"
+SCRIPTS_DIR="$ROOT_DIR/scripts"
+
+# Create temporary directory for test files
+TEMP_DIR=$(mktemp -d)
+trap "rm -rf $TEMP_DIR" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)